repos/system_upgrade/el7toel8/actors/prepareupgradetransaction/actor.py

```diff
@@ -208,54 +206,72 @@ def process(self):
                          ' to proceed the upgrade. Register your system with the'
                          ' subscription-manager tool and attach'
                          ' it to proper SKUs to be able to proceed the upgrade.'))
-            self.produce_error(error)
+            preparetransaction.produce_error(self, error)
             return
 
         # TODO: Find a better place where to run this (perhaps even gate this behind user prompt/question)
-        preparetransaction.call(['/usr/bin/dnf', 'clean', 'all'])
+        # FIXME: fails on insufficient permissions
+        cmd = ['/usr/bin/dnf', 'clean', 'all']
+        _, error = preparetransaction.guard_call(cmd)
+        if error:
+            error.summary = 'Cannot perform dnf cleanup.'
+            preparetransaction.produce_error(self, error)
+            return
 
         # prepare container #
+        # TODO: wrap in one function (create ofs dirs, mount), or even context
+        #       manager (enter -> create dirs, mount; exit -> umount)?
         ofs_info, error = preparetransaction.create_overlayfs_dirs(container_root)
         if not ofs_info:
-            self.produce_error(error)
+            preparetransaction.produce_error(self, error)
             return
 
         error = preparetransaction.mount_overlayfs(ofs_info)
         if error:
-            self.produce_error(error)
+            preparetransaction.produce_error(self, error)
             preparetransaction.remove_overlayfs_dirs(container_root)
             return
 
         # switch EngID to use RHEL 8 subscriptions #
-        prev_rhsm_release = preparetransaction.call(['subscription-manager', 'release'])[0]
+        cmd = ['subscription-manager', 'release']
+        prev_rhsm_release, error = preparetransaction.guard_call(
```

Comment:
I wonder if guard_call couldn't be rather a context manager for more generic usage:
```python
with call_guard(checkers=....):
    do_call()
```